File: ts/exercise-reference/completed/src/prompts.ts
import { ABBYYBankMCP } from "./index.js";

export async function initializePrompts(agent: ABBYYBankMCP) {

  agent.server.registerPrompt(
  "open-bank-account",
  {
			title: 'Open Bank Account',
			description: 'Begin the bank account application process with ABBYY Bank',
			argsSchema: {}
	},
  async () => {
    return {
				messages: [
					{
						role: 'user',
						content: {
							type: 'text',
							text: `I'd like to open a bank account with ABBYY Bank, please.`
						},
					},
				],
      }
   }
)
}

File: py/exercise-reference/completed/prompts.py
def register_prompts(mcp, state):
    @mcp.prompt(title="Open Bank Account", description="Begin the bank account application process with ABBYY Bank")
    def open_bank_account() -> dict:
        return {
            "messages": [
                {
                    "role": "user",
                    "content": {
                        "type": "text",
                        "text": "I'd like to open a bank account with ABBYY Bank, please."
                    }
                }
            ]
        } 
